### PR TITLE
DRILL-6832: Remove the old "unmanaged" external sort

### PR DESCRIPTION
**Pull request by paul-rogers**
When the "managed" external sort was implemented a couple of years back, we retained the original "unmanaged" version out of an abundance of caution. The new version is now battle tested and it is time to retire the original one.

Tests: reran all tests. Adjusted one test where the metrics names for the two sort versions got out of sync.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/BatchGroup.java [hunk 1]
```diff
@@ -18,140 +18,300 @@
...
+      if (writer != null) {
+        spillSet.close(writer);
+        logger.trace("Summary: Wrote {} bytes in {} us to {}", writer.getBytesWritten(), writer.time(TimeUnit.MICROSECONDS), path);
+        writer = null;
+      }
     }
+  }
+
+  protected VectorContainer currentContainer;
+  protected int pointer = 0;
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/BatchGroup.java]: Thanks, looks better now.

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java [hunk 2]
```diff
@@ -286,540 +291,419 @@ public void buildSchema() throws SchemaChangeException {
...
-    stats.setLongStat(Metric.SPILL_COUNT, spillCount);
-    BatchGroup newGroup = new BatchGroup(c1, fs, outputFile, oContext);
-    try (AutoCloseable a = AutoCloseables.all(batchGroupList)) {
-      logger.info("Merging and spilling to {}", outputFile);
-      while ((count = copier.next(targetRecordCount)) > 0) {
-        outputContainer.buildSchema(BatchSchema.SelectionVectorMode.NONE);
-        outputContainer.setRecordCount(count);
-        // note that addBatch also clears the outputContainer
-        newGroup.addBatch(outputContainer);
+    switch (lastKnownOutcome) {
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java]: Ok, thanks for the clarification, I added an appropriate comment under Jira ticket.

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java [hunk 3]
```diff
@@ -286,540 +291,419 @@ public void buildSchema() throws SchemaChangeException {
...
-            .addContext("batchGroups.size", batchGroups.size())
-            .addContext("spilledBatchGroups.size", spilledBatchGroups.size())
-            .addContext("allocated memory", oAllocator.getAllocatedMemory())
-            .addContext("allocator limit", oAllocator.getLimit())
+  /**
+   * Handle a new schema from upstream. The ESB is quite limited in its ability
+   * to handle schema changes.
+   */
+
+  private void setupSchema()  {
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java]: ok, fine.

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java [hunk 4]
```diff
@@ -286,540 +291,419 @@ public void buildSchema() throws SchemaChangeException {
...
+   * we track only the first exception, not potential cascading downstream
+   * exceptions.
+   * <p>
+   * Some Drill code ends up calling close() two or more times. The code
+   * here protects itself from these undesirable semantics.
+   * </p>
+   */
+
+  @Override
+  public void close() {
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java]: Ok, agree. I just didn't know about the battle against warnings and thought it's not a problem to extend ```AutoCloseable``` for nullable fields.

File: common/src/main/java/org/apache/drill/common/AutoCloseables.java [hunk 5]
```diff
@@ -75,6 +76,14 @@ public static void close(AutoCloseable... autoCloseables) throws Exception {
     close(Arrays.asList(autoCloseables));
   }
 
+  public static void closeWithUserException(AutoCloseable... autoCloseables) {
+    try {
+      close(Arrays.asList(autoCloseables));
```
Comment (ihuzenko) [inline common/src/main/java/org/apache/drill/common/AutoCloseables.java]:
```suggestion
      close(autoCloseables);
```

**Comment by ihuzenko** [comment]
@paul-rogers , omg, sometimes I'm so blind:( That last comment from my side was a big mistake and caused StackOverFlow error in our tests with this ```Autocloseable.close(varargs)```. I'm so sorry about that, please revert the last commit.

**Comment by paul-rogers** [comment]
@ihuzenko, I should have spotted that one. Also, I should have done my own test run, but I thought, hey, it is one simple change, what could go wrong? My fault for not actually thinking. Anyway, we should be good now.

**Comment by paul-rogers** [comment]
Rebased on the latest master and reran all unit tests.

**Comment by ihuzenko** [comment]
LGTM, +1. Thanks, @paul-rogers for making the changes.

**Comment by ihuzenko** [comment]
@paul-rogers please squash commits into one.

**Comment by paul-rogers** [comment]
@ihuzenko, thanks much for your review. Squashed commits.

**Pull request closed without merging**